File: src/main/java/com/github/mikn/lavawalker/client/screen/LavaWalkerModMenuIntegration.java
package com.github.mikn.lavawalker.client.screen;

import com.github.mikn.lavawalker.config.LavaWalkerConfig;
import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import me.shedaniel.autoconfig.AutoConfig;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;

@Environment(EnvType.CLIENT)
public class LavaWalkerModMenuIntegration implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return parent -> AutoConfig.getConfigScreen(LavaWalkerConfig.class, parent).get();
    }
}

File: src/main/kotlin/com/github/mikn/lavawalker/client/screen/LavaWalkerModMenuIntegration.kt
package com.github.mikn.lavawalker.client.screen

import com.github.mikn.lavawalker.config.LavaWalkerConfig
import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import me.shedaniel.autoconfig.AutoConfig
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.client.gui.screens.Screen

@Environment(EnvType.CLIENT)
class LavaWalkerModMenuIntegration : ModMenuApi {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        return ConfigScreenFactory { parent: Screen? ->
            AutoConfig.getConfigScreen(
                LavaWalkerConfig::class.java, parent
            ).get()
        }
    }
}